app/views/idv/otp_verification/show.html.erb

```diff
@@ -38,8 +38,9 @@
 
 <%= button_to idv_resend_otp_path,
               method: :post,
-              class: 'usa-button usa-button--outline ico ico-refresh margin-bottom-4' do %>
-  <%= t('links.two_factor_authentication.get_another_code') %>
+              class: 'usa-button usa-button--outline margin-bottom-4' do %>
+  <%= render IconComponent.new(icon: :loop) -%>
+  <% concat(t('links.two_factor_authentication.get_another_code')) %>
```

Comment:
So this was an interesting predicament: The icon and accompanying text here are sensitive to whitespace. This isn't the case with the upstream proposal in https://github.com/uswds/uswds/pull/4493, since it uses `float`. We can't use `float` here, because full-width buttons (a Login.gov-specific button feature) would cause the icon to appear at the far left of the button.

The combination of `-%>` on line 42 and `<% concat` on line 43 avoid whitespace. There are some other ways to do this, though:

- Have them be on the same line
- Use something like `<%= safe_join([ ... ]) %>`

Ultimately I'm also trying to optimize for: What's easiest for future developers to copy/paste as reference and not get wrong.

Reply:
Another option could be to have something like an `IconButtonComponent`, which wouldn't be a far stretch from what we have already with `ButtonComponent`, and could absorb this implementation detail. The problem is that I'm not totally sure how well that'd work with the many variations we'd need to support: `button_to`, `button_tag`, `f.button`, `link_to`, and perhaps others.

Reply:
I should also clarify there's a desire to _not_ include additional class names or child elements to make this work. It could be easily solved if we could e.g. wrap the text in a `<span>`, or change the wrapper to a `display: flex`.

Reply:
I like the idea of an `IconButtonComponent`, or maybe update the button component to have an optional `icon:` param?

Reply:
>I like the idea of an `IconButtonComponent`, or maybe update the button component to have an optional `icon:` param?

It's a bit hastily put together as I approach the end of my day so there may be some mistakes, but how do you feel about the recent round of revisions? The "factory" business feels a bit wonky to me, but otherwise it's got all the upsides that I'd expect from componentizing this behavior.

Reply:
Left some comments here since it felt easier to comment on the impl: https://github.com/18F/identity-idp/pull/5904/files#r798056283